src/Tinfoil/Random.hs

```diff
@@ -20,15 +20,26 @@ import           Data.Word (Word32)
 
 import           P
 
-import qualified System.Entropy as E
 import           System.IO
 
 import           Tinfoil.Data
 import           Tinfoil.Random.Internal
 
 -- | Reads the specified number of bytes from from /dev/urandom.
+--
+-- On OS X this calls a 160-bit Yarrow generator and will block until
+-- the entropy pool has been initialised (should never happen); on
+-- Linux it is a sha1-based generator and will not block regardless of
+-- pool state.
+--
+-- This function incurs some overhead due to GHC.IO buffering; if this
+-- becomes significant (e.g., key generation), using read(3) directly
+-- would make small reads significantly faster.
+--
+-- FIXME: is it worth checking uname and failing on non-linux/darwin here?
 entropy :: Int -> IO Entropy
-entropy = fmap Entropy . E.getEntropy
+entropy n =
+  Entropy <$> withFile "/dev/urandom" ReadMode (\h -> BS.hGet h n)
```

Comment:
> suggests that it might be desirable to check for the specified number of bytes just in case "/dev/urandom" does something unexpected (and potentially really bad).

Yup fair call.

> Also, worth using withBinaryFile in place of withFile there?

Yep I see no downside, probably worth doing in case we ever need to port to windows.